pkg/cache/cache.go

```diff
@@ -158,6 +159,15 @@ var defaultResyncTime = 10 * time.Hour
 
 // New initializes and returns a new Cache.
 func New(config *rest.Config, opts Options) (Cache, error) {
+	httpClient, err := rest.HTTPClientFor(config)
```

Comment:
What is your use-case for this that can not be satisfied by setting a custom `Transport` in the config?

Reply:
Here https://github.com/kubernetes/kubernetes/pull/105490 you can find more info about the upstream decision.
This is one of the problem statements that decision was based on: https://github.com/kubernetes/kubernetes/issues/103225.